### ISSUE TITLE
"Diagnostic Error: [object Object]" when compiling

### ISSUE THREAD

**Issue by evaera** [roblox-ts/roblox-ts#146]
Attempting to run `rbxtsc` when the `out` folder already exists spams my console with this:
```
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
Diagnostic Error: [object Object]
```

If I delete `out`, then it'll work, but breaks again due to the folder being present once more. Only happens with this project, unsure what triggers it.

Repro: 
[rbx-resources.zip](https://github.com/roblox-ts/roblox-ts/files/2623338/rbx-resources.zip)

**Comment by Vorlias** [comment]
https://github.com/roblox-ts/roblox-ts/blob/master/src/class/Compiler.ts#L379

`getMessageText()` returns both a `ts.DiagnosticMessageChain` as well as a `string`.
In this case, it seems the error message is a `ts.DiagnosticMessageChain`. I've managed to fix it to get the real error message like so:

```ts
const messageText = diagnostic.getMessageText();
if (messageText instanceof ts.DiagnosticMessageChain) {
	console.log("%s%s %s", prefix, red("Diagnostic Error:"), messageText.getMessageText());
} else {
	console.log("%s%s %s", prefix, red("Diagnostic Error:"), diagnostic.getMessageText());
}
```

The actual error is that it's overwriting an input file.

`Diagnostic Error: Cannot write file '[truncated]/out/index.d.ts' because it would overwrite input file.`

**Comment by evaera** [comment]
Error message display problem fixed in 1c7179c6220ae7781b11c0d0d116023de8c021d5
